prow/config/config.go

```diff
@@ -326,8 +338,34 @@ func parseConfig(c *Config) error {
 		return fmt.Errorf("validating plank config: %v", err)
 	}
 
-	if err := ValidateController(&c.JenkinsOperator.Controller); err != nil {
-		return fmt.Errorf("validating jenkins-operator config: %v", err)
+	if c.JenkinsOperator != nil && len(c.JenkinsOperators) > 0 {
+		return fmt.Errorf("invalid config: use just one out of jenkins_operator and jenkins_operators")
+	}
+	if c.JenkinsOperator != nil {
+		if err := ValidateController(&c.JenkinsOperator.Controller); err != nil {
+			return fmt.Errorf("validating jenkins-operator config: %v", err)
+		}
+		if c.JenkinsOperator.LabelSelectorString != "" {
+			return errors.New("use jenkins_operators for sharding prow between Jenkins masters")
+		}
+	}
+	for i := range c.JenkinsOperators {
+		config := c.JenkinsOperators[i]
+		if err := ValidateController(&config.Controller); err != nil {
+			return fmt.Errorf("validating jenkins_operators config: %v", err)
+		}
+		sel, err := labels.Parse(config.LabelSelectorString)
+		if err != nil {
+			return fmt.Errorf("invalid jenkins_operators.label_selector option: %v", err)
+		}
+		c.JenkinsOperators[i].LabelSelector = sel
+		// TODO: Invalidate overlapping selectors
+	}
+	// Be nice to folks and stay backwards compatible
+	if c.JenkinsOperator != nil {
+		logrus.Warning("jenkins_operator is deprecated, please switch to the jenkins_operators option!")
+		c.JenkinsOperators = make([]JenkinsOperator, 1)
+		c.JenkinsOperators = append(c.JenkinsOperators, *c.JenkinsOperator)
```

Comment:
The `LabelSelector` and `LabelSelectorString` fields need to be set before adding `c.JenkinsOperator` to `c.JenkinsOperators` because these fields are assumed to be valid elsewhere.

Reply:
It's invalid using `label_selector` when you use `jenkins_operator` (because it is assumed you already select everything, if you need sharding you have to use `jenkins_operators`). This `append` is only convenient for maintaining backwards-comaptibility.

Reply:
I think my change to your other comment addresses this. If `jenkins_operator` is used, the controller config will always be the same across replicas. I have also added a comment in `LabelSelectorString` to denote that it's invalid to use with a single jenkins-operator config (whether it is `jenkins_operator` or `jenkins_operators` with a single jenkins config).